src/Database/Type.php

```diff
@@ -132,28 +99,23 @@ public static function buildAll()
      * Returns a Type object capable of converting a type identified by $name
      *
      * @param string $name The type identifier you want to set.
-     * @param \Cake\Database\Type $instance The type instance you want to set.
+     * @param \Cake\Database\TypeInterface $instance The type instance you want to set.
      * @return void
      */
-    public static function set($name, Type $instance)
+    public static function set($name, TypeInterface $instance)
     {
         static::$_builtTypes[$name] = $instance;
     }
 
     /**
      * Registers a new type identifier and maps it to a fully namespaced classname,
-     * If called with no arguments it will return current types map array
      * If $className is omitted it will return mapped class for $type
      *
-     * Deprecated: The usage of $type as \Cake\Database\Type[] is deprecated. Please always use string[] if you pass an array
-     * as first argument.
-     *
-     * @param string|string[]|\Cake\Database\Type[]|null $type If string name of type to map, if array list of arrays to be mapped
-     * @param string|\Cake\Database\Type|null $className The classname or object instance of it to register.
-     * @return array|string|null If $type is null then array with current map, if $className is null string
-     * configured class name for give $type, null otherwise
+     * @param string|string[] $type If string name of type to map, if array list of arrays to be mapped.
+     * @param string|\Cake\Database\TypeInterface|null $className The classname or object instance of it to register.
+     * @return string|null If $className is null string configured class name for give $type, null otherwise
      */
-    public static function map($type = null, $className = null)
+    public static function map($type, $className = null)
```

Comment:
If it can't be null anymore, we should remove the return part `if ($type === null)`.
Also, we should maybe move the array input part out of this method. This is weird, that this then assigns directly, not using the 2nd argument. An own method mapAll() or something can be used here maybe?

Reply:
And if this is a true setter, then

    if ($className === null) {
             return isset(static::$_types[$type]) ? static::$_types[$type] : null;
         }

is also obsolete, the $className should then not be nullable anymore, either.